### ISSUE TITLE
Globally disable analyzer warnings in bootstrapping stage 2 build

### ISSUE THREAD

**Issue by mthalman** [dotnet/source-build#3427]
One of the frequent causes of failures in the bootstrapping stage 2 build is due to analyzer warnings (which are treated as errors). Recent examples of this are https://github.com/dotnet/installer/pull/16269 and https://github.com/dotnet/installer/pull/16268. Failing the build on these types of errors is problematic:
* It blocks the release.
* It prevents visibility into potentially other, more serious, downstream issues.
* It's time-consuming to investigate, fix, and verify.

Failing on these errors also doesn't provide much value either since they really don't affect the functionality of the product.

We should consider globally disabling all analyzer warnings/errors for all repos built in source-build.

**Comment by MichaelSimons** [comment]
I don't think this is going to reduce any source-build work rather it is just going to delay when the errors appear.  If you suppress warnings as errors in stage 2, you will encounter then in the post release PRs that update the SDK to the latest version.

**Comment by MichaelSimons** [comment]
[Triage] We are now running stage 2 legs are part of PR validation.  We therefore catch and fix any breaks up front.  Closing.
